src/FromToStateTransition.php

```diff
@@ -0,0 +1,77 @@
+<?php
+/**
+ * This file is part of the php-state project.
+ *
+ * (c) Yannick Voyer <star.yvoyer@gmail.com> (http://github.com/yvoyer)
+ */
+
+namespace Star\Component\State;
+
+use Webmozart\Assert\Assert;
+
+final class FromToStateTransition implements StateTransition
```

Comment:
* Add `OneToManyStateTransition`
* Add `OneToOneStateTransition`
* Add `ManyToOneStateTransition`
* Add `ManyToManyTransition`
* Add `ClosureTransition`
* Add `NotAllowedTransition` that is used internally to blacklist.